.github/workflows/linux.yml

```
@@ -25,8 +25,26 @@ jobs:
       run: >
        sudo apt-get update && sudo apt install libx11-dev libxext-dev libwayland-dev libfuse2 clang build-essential
 
+    - name: Cache CMake dependency source code
+      uses: actions/cache@v4
+      env:
+          cache-name: ${{ runner.os }}-sdl-cache-cmake-dependency-sources
+      with:
+          path: | 
+            ${{github.workspace}}/build
+          key: ${{ env.cache-name }}-${{ hashFiles('**/CMakeLists.txt', 'cmake/**') }}
+          restore-keys: |
+            ${{ env.cache-name }}-
+
+    - name: Cache CMake dependency build objects
+      uses: hendrikmuhs/ccache-action@v1.2.14
+      env:
+          cache-name: ${{ runner.os }}-sdl-cache-cmake-dependency-builds
+      with:
+        key: ${{ env.cache-name }}-
```

Using `append-timestamp` as `false` will avoid the creation of unnecessary multiples caches for ccache, and it's better to remove the '-' from `${{ env.cache-name }}-` because the action always append the '-'.
This is also required on `linux-qt.yml`

```suggestion
      with:
        append-timestamp: false
        key: ${{ env.cache-name }}
```